packages/sdk/src/assets/open-edition-nft.ts

```diff
@@ -1,20 +1,73 @@
-import {Hex, pad, toHex} from 'viem'
+import {Address, Hex, pad, toHex} from 'viem'
 
 import {GAS_CONTROLLER} from '../constants/gas-controllers'
 import {HolographOpenEditionERC721Contract} from '../contracts'
 import {queryTokenIdFromReceipt} from '../utils/decoders'
 import {HolographVersion, MintConfig, WriteContractOptions} from '../utils/types'
-import {NFT} from './nft'
-import {CreateNFT} from './nft.validation'
+import {CreateOpenEditionNFT, HolographOpenEditionNFTMetadata, validate} from './nft.validation'
+import {RequireMintedToken} from '../utils/decorators'
+import {getParsedTokenId} from '../utils/transformers'
+import {
+  HolographOpenEditionERC721ContractV1,
+  HolographOpenEditionERC721ContractV2,
+} from './holograph-open-edition-erc721-contract'
+import {NotMintedNFTError} from '../errors'
+
+export class OpenEditionNFT {
+  public contract: HolographOpenEditionERC721ContractV1 | HolographOpenEditionERC721ContractV2
+  public isMinted: boolean
+  public txHash?: string
+
+  protected _tokenId?: string // Decimal tokenId string
+  private _metadata!: HolographOpenEditionNFTMetadata
 
-export class OpenEditionNFT extends NFT {
   private holographOpenEditionERC721: HolographOpenEditionERC721Contract
 
-  constructor({contract, metadata, version = HolographVersion.V2}: CreateNFT) {
-    super({contract, metadata, version})
+  constructor({contract, version = HolographVersion.V2}: CreateOpenEditionNFT) {
+    this.contract = validate.openEditionContract.parse(contract)
 
-    const holographOpenEditionERC721 = new HolographOpenEditionERC721Contract(contract.contractAddress!, version)
-    this.holographOpenEditionERC721 = holographOpenEditionERC721
+    this.holographOpenEditionERC721 = new HolographOpenEditionERC721Contract(contract.contractAddress!, version)
+    this.isMinted = false
+  }
+
+  @RequireMintedToken()
+  get metadata() {
+    return this._metadata
+  }
+
+  @RequireMintedToken()
```

Comment:
Could you let me know why we need this here? 

I think if I just instantiated the class, I should be able to get my NFT name even though I haven't minted yet

Reply:
The NFT Open Edition (OE) name follows this pattern: CollectionName + TokenID. Since the TokenID is only generated after the token is minted, we can't determine the name beforehand.